Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+TypeSignature.swift

```diff
@@ -16,130 +16,313 @@ import SymbolKit
+                }
+                
+                switch fragment.kind {
+                case .typeIdentifier:
+                    // Accumulate all of the identifier tokens' spelling.
+                    accumulated.append(contentsOf: fragment.spelling.utf8)
+                    
+                case .text: // In Swift, only `text` tokens contains whitespace
+                    var spelling = fragment.spelling.utf8[...]
+                    // If the type spelling is a parameter, if often starts with a leading ":" that's not part of the type name.
+                    if accumulated.isEmpty, spelling.first == colon {
+                        _ = spelling.removeFirst()
+                    }
+                    
+                    // Ignore whitespace in text tokens. Here we use a loop instead of `filter` to avoid a potential temporary allocation.
+                    for char in spelling where char != space {
+                        accumulated.append(char)
+                    }
+                    
+                default:
+                    continue
+                }
+            } else {
+                switch fragment.kind {
+                case .identifier where preciseIdentifier != nil,
+                     .typeIdentifier,
+                     .text:
+                    let spelling = fragment.spelling.utf8
+                    
+                    // Ignore whitespace. Here we use a loop instead of `filter` to avoid a potential temporary allocation.
+                    for char in spelling where char != space {
+                        accumulated.append(char)
+                    }
+                    
+                default:
+                    continue
+                }
             }
         }
         
-        accumulated.removeAll(where: \.isWhitespace)
-        
-        if accumulated.first == ":" {
-            _ = accumulated.removeFirst()
+        // Check if the type names are wrapped in redundant parenthesis and remove them
+        if accumulated.first == openParen, accumulated.last == closeParen, !accumulated[...].isTuple() {
+            // In case there are multiple
+            // Use a temporary slice until all the layers of redundant parenthesis have been removed.
+            var temp = accumulated[...]
+            
+            repeat {
+                temp = temp.dropFirst().dropLast()
+            } while temp.first == openParen && temp.last == closeParen && !temp.isTuple()
+            
+            // Adjust the markers so that they align with the expected characters
+            let difference = (accumulated.count - temp.count) / 2
+            
+            accumulated = .init(temp)
+            
+            for index in markers.indices {
+                markers[index] -= difference
+                
+                assert(accumulated[markers[index]] == uppercaseA || accumulated[markers[index]] == uppercaseD  || accumulated[markers[index]] == uppercaseO, """
```

Comment:
The `allSatisfy` assertion below checks that the "final" accumulated string has correctly aligned markers.

This assertion checks that the code that removes redundant leading `(` and trailing `)` also updates the markers to continue to be aligned. If there's a bug in this code it could make the investigation easier with the assertion closer to the code that caused it.